doc/DifferentiableManifoldRecipe.md

```diff
@@ -0,0 +1,180 @@
+   automatically-generated derivatives are exactly the tangent vectors
+   corresponding to the local coordinate system you specify.
+
+Note: We want to make the recipe simpler. See
+[issue #45](https://github.com/borglab/SwiftFusion/issues/45).
+
+## Steps
+
+### 1. Choose a global coordinate system.
+
+Create a `struct` whose values represent points on the manifold. Every point on
+the manifold must be representable by at least one `struct` value. Every
+`struct` value must represent at most one point on the manifold.
+
+For example, both of these `struct`s are acceptable global coordinate systems
+for `SO(2)`, the manifold of 2D rotations:
+
+```swift
+struct SinCosGlobalCoordinate {
+  /// The cosine of the angle of the rotation.
+  var c: Float
+
+  /// The sine of the angle of the rotation.
+  var s: Float
+}
+
+struct AngleGlobalCoordinate {
+  /// The angle of the rotation.
+  var theta: Float
+}
+```
+
+### 2. Choose the local coordinate type.
+
+Choose an `n`-dimensional vector type to represent to represent the manifold's
+local coordinates.
+
+For example, `Float` and `Vector1` are both acceptable types for `SO(2)` local
+coordinates.
+
+### 3. Conform to the ManifoldCoordinate protocol.
+
+The coordinate `struct` you chose in step 1 should conform to the
+`ManifoldCoordinate` protocol, which has these requirements:
+
+```swift
+public protocol ManifoldCoordinate: Differentiable {
+  /// The local coordinate type of the manifold.
+  ///
+  /// This is the `TangentVector` of the `Manifold` wrapper type.
+  ///
+  /// Note that this is not the same type as `Self.TangentVector`.
+  associatedtype LocalCoordinate: EuclideanVectorSpace
+
+  /// Diffeomorphism between a neigborhood of `Localcoordinate.zero` and `Self`.
+  ///
+  /// Satisfies the following properties:
+  /// - `retract(LocalCoordinate.zero) == self`
+  /// - There exists an open set `B` around `LocalCoordinate.zero` such that
+  ///   `localCoordinate(retract(b)) == b` for all `b \in B`.
+  @differentiable(wrt: local)
+  func retract(_ local: LocalCoordinate) -> Self
+
+  /// Inverse of `retract`.
+  ///
+  /// Satisfies the following properties:
+  /// - `localCoordinate(self) == LocalCoordinate.zero`
+  /// - There exists an open set `B` around `self` such that `localCoordinate(retract(b)) == b` for all
+  ///   `b \in B`.
+  @differentiable(wrt: global)
+  func localCoordinate(_ global: Self) -> LocalCoordinate
+}
+```
+
+(See the code for more detailed requirements and documentation).
+
+The implementations of `global` and `local` determine how the manifold's global
+and local coordinates are related, and they therefore determine the behavior of
+the manifold's tangent vectors during differentiation.
+
+Note: These are also known as the "exponential map" and "logarithm" in
+differential geometry.
+
+### Step 4. Create the manifold type.
+
+Create a struct with a stored property and initalizer as follows:
+```
+var coordinateStorage: ${your coordinate type}
```

Comment:
Don't think you should be using gyb syntax here 😉 

It's unfortunate that we don't have a way to embed typographic styles into code snippets in markdown, or we could use italics.  As it stands, I would just put "where `X` is your coordinate type” in the prose, and use “`X`”.